cluster-operator/src/main/java/io/strimzi/operator/cluster/model/AbstractModel.java

```diff
@@ -1080,4 +1092,24 @@ public static String clusterCaCertSecretName(String cluster)  {
     public static String clusterCaKeySecretName(String cluster)  {
         return cluster + "-cluster-ca";
     }
+
+    protected static Map<String, String> mergeAnnotations(Map<String, String> internal, Map<String, String> template) {
+        Map<String, String> merged = new HashMap<>();
+
+        if (internal != null) {
+            merged.putAll(internal);
+        }
+
+        if (template != null) {
+            for (String key : template.keySet()) {
```

Comment:
Maybe we can use `streams` in this way?

```java
boolean isPresent = template.keySet().stream()
        .filter(k -> k.contains("strimzi.io"))
        .findFirst().isPresent();
```

but change is not needed, if you like using the for cycle.

Reply:
Do you think it's faster using streams? I normally find the `for` loop easier to read. But I don't have problem changing it to streams.

Reply:
No worries let's stay with for. Merge! ;)